app/test_main.py

```
@@ -1,3 +1,115 @@
+        (28, 28, [3, 2]),
+    ])
+    def test_ages_at_and_after_second_threshold(
+            self, cat_age: int, dog_age: int, expected: list
+    ) -> None:
+        assert get_human_age(cat_age, dog_age) == expected
+
+    @pytest.mark.parametrize("cat_age,dog_age,expected", [
+        (15, 20, [1, 1]),
+        (20, 15, [1, 1]),
+        (30, 25, [3, 2]),
+        (25, 30, [2, 3]),
+        (28, 29, [3, 3]),
+        (29, 28, [3, 2]),
+    ])
+    def test_different_cat_dog_ages(
+            self, cat_age: int, dog_age: int, expected: list
+    ) -> None:
+        assert get_human_age(cat_age, dog_age) == expected
+
+    @pytest.mark.parametrize("cat_age,dog_age,expected", [
+        (50, 50, [8, 7]),
+        (100, 100, [21, 17]),
+        (200, 200, [46, 37]),
+        (1000, 1000, [246, 197]),
+    ])
+    def test_large_ages(
+            self, cat_age: int, dog_age: int, expected: list
+    ) -> None:
+        assert get_human_age(cat_age, dog_age) == expected
+
+    @pytest.mark.parametrize("cat_age,expected_cat", [
+        (15, 1),
+        (24, 2),
+        (28, 3),
+        (32, 4),
+    ])
+    def test_cat_specific_calculation(
+            self, cat_age: int, expected_cat: int
+    ) -> None:
+        result = get_human_age(cat_age, 0)
+        assert result[0] == expected_cat
+
+    @pytest.mark.parametrize("dog_age,expected_dog", [
+        (15, 1),
+        (24, 2),
+        (29, 3),
+        (34, 4),
+    ])
+    def test_dog_specific_calculation(
+            self, dog_age: int, expected_dog: int
+    ) -> None:
+        result = get_human_age(0, dog_age)
+        assert result[1] == expected_dog
+
+    def test_negative_ages_behavior(self) -> None:
+        assert get_human_age(-1, 0) == [0, 0]
+        assert get_human_age(0, -1) == [0, 0]
+        assert get_human_age(-5, -10) == [0, 0]
+
+    def test_float_inputs_work(self) -> None:
+        assert get_human_age(15.5, 20.5) == [1, 1]
+        assert get_human_age(15.0, 20.0) == [1, 1]
```

Ambiguous float policy: test_float_inputs_work currently asserts that float inputs are accepted and return valid results. This conflicts with the checklist/description that inputs are integers and that incorrect types should raise an exception. The checklist explicitly requires: "* if the function receives an incorrect type of data it raises the correct exception." Decide whether floats are allowed or not and update tests accordingly (either expect an exception for floats or assert integer outputs for float inputs and document the decision)  .